mmv1/third_party/terraform/services/bigtable/resource_bigtable_authorized_view.go

```diff
@@ -0,0 +1,483 @@
+			},
+
+			"instance_name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: tpgresource.CompareResourceNames,
+				Description:      `The name of the Bigtable instance in which the authorized view belongs.`,
+			},
+
+			"table_name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: tpgresource.CompareResourceNames,
+				Description:      `The name of the Bigtable table in which the authorized view belongs.`,
+			},
+
+			"deletion_protection": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringInSlice([]string{"PROTECTED", "UNPROTECTED"}, false),
+				Description:  `A field to make the authorized view protected against data loss i.e. when set to PROTECTED, deleting the authorized view, the table containing the authorized view, and the instance containing the authorized view would be prohibited. If not provided, currently deletion protection will be set to UNPROTECTED as it is the API default value.`,
+			},
+
+			"subset_view": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				MaxItems:    1,
+				Description: `An AuthorizedView permitting access to an explicit subset of a Table.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"row_prefixes": {
+							Type:        schema.TypeSet,
+							Optional:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Description: `Base64-encoded row prefixes to be included in the authorized view. To provide access to all rows, include the empty string as a prefix ("").`,
+						},
+						"family_subsets": {
+							Type:        schema.TypeSet,
+							Optional:    true,
+							Description: `Subsets of column families to be included in the authorized view.`,
+							Elem:        familySubsetSchemaElem,
+						},
+					},
+				},
+			},
+		},
+		UseJSONNumber: true,
+	}
+}
+
+func resourceBigtableAuthorizedViewCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+	userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+
+	project, err := tpgresource.GetProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	instanceName := tpgresource.GetResourceNameFromSelfLink(d.Get("instance_name").(string))
+	c, err := config.BigTableClientFactory(userAgent).NewAdminClient(project, instanceName)
+	if err != nil {
+		return fmt.Errorf("Error starting admin client. %s", err)
+	}
+	if err := d.Set("instance_name", instanceName); err != nil {
+		return fmt.Errorf("Error setting instance_name: %s", err)
+	}
+
+	defer c.Close()
+
+	authorizedViewId := d.Get("name").(string)
+	tableId := d.Get("table_name").(string)
+	authorizedViewConf := bigtable.AuthorizedViewConf{
```

Comment:
Apologies, I thought I left a comment but it must not have sent.
The build is failing with `bigtable.AuthorizedViewConf` being undefined.

I believe we will have to update the `cloud.google.com/go/bigtable` dependency

yaqs/8444818848342867968 should have a note that explains how to do that in our `go.mod`

Reply:
Sorry it took a while for us to merge the `cloud.google.com/go/bigtable` dependency for this PR. Now the dependency is finally merged, and the `go.mod` is updated accordingly.

Can you please start review this PR? Thanks!

Reply:
We should update the [go.sum](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/go.sum) as gets generated as well. You should just be able to copy and paste the result after running `go get cloud.google.com/go/bigtable` in the provider

Reply:
Thanks for the info! Done.